ui/guest/src/index.tsx

```diff
@@ -101,6 +103,36 @@ export function initExperienceBuilder(props: ExperienceBuilderProps) {
 /** @deprecated Use `initExperienceBuilder` instead. */
 export const initInContextEditing = initExperienceBuilder;
 
+export const guestCheckIn$ = guestCheckInInternal$.pipe(
+  filter((checkedIn) => checkedIn),
+  take(1)
+);
+
 export { elementRegistry, iceRegistry, contentController, fromTopic, post };
 
 typeof document !== 'undefined' && setTimeout(() => document?.dispatchEvent(new CustomEvent(xbLoadedEvent)));
+
+export function registerElements(container: HTMLElement) {
+  if (container) {
+    container.querySelectorAll('[data-craftercms-model-id]').forEach((element) => {
+      elementRegistry.register({
+        element: element as HTMLElement,
+        path: element.getAttribute('data-craftercms-model-path'),
+        modelId: element.getAttribute('data-craftercms-model-id'),
+        fieldId: element.getAttribute('data-craftercms-field-id'),
+        index: element.getAttribute('data-craftercms-index'),
+        label: element.getAttribute('data-craftercms-label')
+      });
+    });
+  }
+}
+
+export function deregisterElements(container: HTMLElement) {
+  if (container) {
+    const articleElements = container.querySelectorAll('[data-craftercms-model-id]');
+    articleElements.forEach((element) => {
```

Comment:
These aren't articles here...
```suggestion
    const elements = container.querySelectorAll('[data-craftercms-model-id]');
    elements.forEach((element) => {
```